x/staking/types/delegation.go

```diff
@@ -307,6 +303,7 @@ func UnmarshalRED(cdc *codec.Codec, value []byte) (red Redelegation, err error)
 }
 
 // nolint
+// TODO: this is less efficient than comparing struct fields
```

Comment:
yeah but it simplifies the code - and this is only used within testing... we mention this in a comment above here just so it's clear we don't need to optimize this